awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/AwsMetricAttributeGenerator.java

```diff
@@ -392,6 +417,97 @@ private static void setRemoteResourceTypeAndIdentifier(SpanData span, Attributes
     }
   }
 
+  /**
+   * RemoteResourceIdentifier is populated with rule <code>
+   *     ^[db.name|]?{address} [|{port}]?
+   * </code>
+   *
+   * <pre>
+   * {address} attribute is retrieved in priority order:
+   * - {@link SemanticAttributes#SERVER_ADDRESS},
+   * - {@link SemanticAttributes#NET_PEER_NAME},
+   * - {@link SemanticAttributes#SERVER_SOCKET_ADDRESS}
+   * - {@link SemanticAttributes#DB_CONNECTION_STRING}-Hostname
+   * </pre>
+   *
+   * <pre>
+   * {port} attribute is retrieved in priority order:
+   * - {@link SemanticAttributes#SERVER_PORT},
+   * - {@link SemanticAttributes#NET_PEER_PORT},
+   * - {@link SemanticAttributes#SERVER_SOCKET_PORT}
+   * - {@link SemanticAttributes#DB_CONNECTION_STRING}-Port
+   * </pre>
+   *
+   * If address is not present, neither RemoteResourceType nor RemoteResourceIdentifier will be
+   * provided. If any attribute contains | or ^ , they will be replaced with ^| or ^^
+   */
+  private static Optional<String> getDBResourceIdentifier(SpanData span) {
```

Comment:
Not blocking: Generally your terminology is confused, advise rename:
* `getDBResourceIdentifier` to `getDbConnection`
* `dbServerAddress` to `dbConnection` 
* `buildDBResourceIdentifier` to `buildDbConnection`

Reply:
yes. I like this suggestion that make it in a consistent way. thx